index.js

```diff
@@ -68,20 +69,37 @@ WinstonDynamoDB.prototype.createUploadInterval = function () {
 WinstonDynamoDB.prototype.add = function (log) {
     debug('add log to queue', log);
 
-    if (!isEmpty(log.message) || isError(log.message)) {
-        this.logEvents.push({
-            message: this.formatMessage(log),
-            timestamp: process.hrtime.bigint(),
-            rawMessage: log
-        });
-    }
+    const { message: originalMessage } = log;
 
-    // When we reach maximum amount of items in batch, reschedule
-    if (this.logEvents.length >= dynamodbIntegration.MAX_BATCH_ITEM_NUM) {
-        debug('Max items for batch reached - submitting and rescheduling interval');
-        clearInterval(this.intervalId);
-        this.createUploadInterval();
-        this.submit();
+    if (originalMessage.length <= maxMessageLength) {
+        if (!isEmpty(originalMessage) || isError(originalMessage)) {
+            this.logEvents.push({
+                message: this.formatMessage(log),
+                timestamp: process.hrtime.bigint(),
+                rawMessage: log
+            });
+        }
+
+        if (this.logEvents.length >= dynamodbIntegration.MAX_BATCH_ITEM_NUM) {
+            debug('Max items for batch reached - submitting and rescheduling interval');
+            clearInterval(this.intervalId);
+            this.createUploadInterval();
+            this.submit();
+        }
+    } else {
+        for (let i = 0; i < originalMessage.length; i += maxMessageLength) {
+            let currentMessageSlice = originalMessage.slice(i, i + maxMessageLength);
+            this.logEvents.push({
+                message: this.formatMessage({...log, message: currentMessageSlice}),
+                timestamp: process.hrtime.bigint(),
+                rawMessage: log
+            });
+
+            debug(`Send each slice individually right away. current slice number:  ${(i / maxMessageLength) + 1}`);
+            clearInterval(this.intervalId);
+            this.createUploadInterval();
+            this.submit();
```

Comment:
We don't need to send each slice right away. It should take part in the usual manner - either the batch is sent every 500ms (uploadRate), or if we reached more than the maximum number of events then send a batch immediately.

The code would look nicer, and would be more correct, as follows:

```
if (not empty and not error) {
  Add events - split events if necessary (this could be done in a single `for` loop. No need to split up the code with `length <= maxMessageLength`)
}

if (number of events is more than batch max) {
  immediately send
}
```

Reply:
Alright, so let's leave it as it is. It should immediately send each part as its own message to DDB without waiting for a batch. So, the worst case here is that if a message is extremely long we might get throttled by DDB.

But let's see if this is actually something that'll happen or not